spring-cloud-sleuth-samples/spring-cloud-sleuth-sample-test-core/src/main/java/tools/RequestSendingRunnable.java

```diff
@@ -50,9 +51,9 @@ public void run() {
 		log.info("Received the following response [{}]", responseEntity);
 	}
 
-	private RequestEntity requestWithTraceId(String traceId) {
+	private RequestEntity requestWithTraceId(Long traceId) {
 		HttpHeaders headers = new HttpHeaders();
-		headers.add(Trace.TRACE_ID_NAME, traceId);
+		headers.add(Trace.TRACE_ID_NAME, LongUtils.toString(traceId));
```

Comment:
another option would just to be explicit. ex hexEncode hexDecode.. there are existing utilities for that. ex you don't need to name a class when you don't declare one.